src/main/webapp/assets/modules/common/model/jobseekerDashboardChallenge.js

```diff
@@ -0,0 +1,92 @@
+techlooper.filter("jobseekerDashboardChallenge", function ($filter, $translate, utils, jsonValue) {
+  return function (input, type) {
+    if (!input || input.$isRich) return input;
+
+    var challenge = input;
+
+    challenge.recalculate = function () {
+      challenge.$asciiChallengeName = utils.toAscii(challenge.challengeName);
+      challenge.recalculatePhase();
+      challenge.recalculateCriteria();
+      challenge.recalculateAward();
+      challenge.recalculateSubmissions();
+    };
+
+    challenge.recalculatePhase = function () {
+      var currentPhaseLowerCase = challenge.currentPhase.toLowerCase();
+      challenge.$currentPhaseLowerCase = currentPhaseLowerCase;
+      challenge.$currentPhaseTitle = $filter("translate")(currentPhaseLowerCase).toUpperCase();
+
+      challenge.$submissionMdate = moment(challenge.submissionDate, jsonValue.dateFormat);
+
+      var finishDaysLeft = utils.toNow(challenge.$submissionMdate);
+      challenge.$daysLeftTitle = finishDaysLeft > 0 ? $filter("translate")("daysLeft", {days: finishDaysLeft}) :
+        $filter("translate")("finishedOn", {date: challenge.submissionDate});
+
+      if (challenge.disqualified == true) {
+        challenge.$currentPhaseLowerCase = "disqualified";
+        challenge.$currentPhaseDescTitle = $filter("translate")("inPhase", {phase: challenge.$currentPhaseTitle});
+        challenge.$currentPhaseTitle = $filter("translate")("disqualified").toUpperCase();
+      }
+      else {
+        challenge.$currentPhaseSubmissionDate = moment(challenge.currentPhaseSubmissionDate, jsonValue.dateFormat);
+        var submitDaysLeft = Math.max(utils.toNow(challenge.$currentPhaseSubmissionDate), 0);
+        challenge.$currentPhaseDescTitle = finishDaysLeft <= 0 ?  undefined : $filter("translate")("daysLeftToSubmit", {days: submitDaysLeft});
+      }
+    };
+
+    challenge.recalculateSubmissions = function () {
+      _.each(challenge.submissions, function (submission) {
+        submission.$submissionURL = /^https?:\/\//.test(submission.submissionURL) ? submission.submissionURL : "http://" + submission.submissionURL;
+        submission.submissionPhase && (submission.$phaseTitle = $filter("translate")(submission.submissionPhase.toLowerCase()));
+      });
+    };
+
+    challenge.recalculateAward = function () {
+      challenge.$awardLowercase = "price-" + challenge.rank;
+      $.isNumeric(challenge.prize) && (challenge.$awardPrizeTitle = "$" + $filter("number")(challenge.prize));
```

Comment:
Line is too long.<br>Expected an assignment or function call and instead saw an expression.